lib/unleash/metrics_reporter.rb

```diff
@@ -33,16 +35,28 @@ def generate_report
       report
     end
 
-    def send
-      Unleash.logger.debug "send() Report"
+    def post
+      Unleash.logger.debug "post() Report"
+
+      if bucket_empty? && (Time.now - self.last_time < LONGEST_WITHOUT_A_REPORT) # and last time is less then 10 minutes...
+        Unleash.logger.debug "Report not posted to server, as it would have been empty. (and has been empty for up to 10 min)"
+
+        return
```

Comment:
If I'm reading this correctly, the early return here means that self.last_time won't get updated in this call. If that's the case, the logs will get thumped with the warning message until a report actually gets posted. Is that the behavior we want here or is it better to update the time just prior to this so that this warning only gets raised every 10 minutes?

Reply:
That's correct. It's a debug log statement, so it shouldn't be visible for more users anyhow, unless they more or less ask for it.

Given that it is a debug log statement, I don't think we should delay printing it out.